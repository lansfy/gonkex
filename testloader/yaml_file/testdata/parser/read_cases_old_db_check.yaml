- name: loader MUST substitute dbQueryArgs/dbResponseArgs/dbResponse to obsolete dbRequests
  method: GET
  path: /some/path
  response:
    200: "aaabbb"
  dbQuery: "SELECT {{ .foo }}"
  dbResponse:
    - "{{ .bar }} item1"
    - "{{ .bar }} item2"
  cases:
    - name: make variable substitution
      dbQueryArgs:
        foo: value1
      dbResponseArgs:
        bar: value2
    - name: redefine db response
      dbQueryArgs:
        foo: value22
      dbResponse:
        - "otheritem1 {{ .bar }}"
        - otheritem2
        - otheritem3
