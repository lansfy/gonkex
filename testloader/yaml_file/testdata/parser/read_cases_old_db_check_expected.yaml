- GetName: 'loader MUST substitute dbQueryArgs/dbResponseArgs/dbResponse to obsolete dbRequests #1 (make variable substitution)'
  GetMethod: GET
  Path: /some/path
  GetResponses:
    200: "aaabbb"
  GetDatabaseChecks:
    - DbQueryString: "SELECT value1"
      DbResponseJson:
        - value2 item1
        - value2 item2
  GetFileName: testdata/parser/read_cases_old_db_check.yaml
  GetLineNumber: 1
  OneOfCase: true
  FirstTestInFile: true

- GetName: 'loader MUST substitute dbQueryArgs/dbResponseArgs/dbResponse to obsolete dbRequests #2 (redefine db response)'
  GetMethod: GET
  Path: /some/path
  GetResponses:
    200: "aaabbb"
  GetDatabaseChecks:
    - DbQueryString: "SELECT value22"
      DbResponseJson:
        - "otheritem1 {{ .bar }}" # no substitution here
        - otheritem2
        - otheritem3
  GetFileName: testdata/parser/read_cases_old_db_check.yaml
  GetLineNumber: 1
  OneOfCase: true
  LastTestInFile: true
